[Style] Convert block-step-size property to use strong style types

weinig opened the pull request:
#### bae69f8f37e9a6276bf30f509470dfd4bb20ba72
<pre>
[Style] Convert block-step-size property to use strong style types
<a href="https://bugs.webkit.org/show_bug.cgi?id=296335">https://bugs.webkit.org/show_bug.cgi?id=296335</a>

Reviewed by NOBODY (OOPS!).

Convert `block-step-size` property to a strong style type.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/rendering/RenderBlock.cpp:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleExtractorSerializer.h:
* Source/WebCore/style/values/rhythm/StyleBlockStepSize.cpp: Added.
* Source/WebCore/style/values/rhythm/StyleBlockStepSize.h: Added.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bae69f8f37e9a6276bf30f509470dfd4bb20ba72

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112985 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32720 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23198 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119189 "Built successfully") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/63532 "Failed to checkout and rebase branch from PR 48380") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/114947 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33372 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41283 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85985 "Found 4 new test failures: imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/computed-keyframes-shorthands.html (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/63532 "Failed to checkout and rebase branch from PR 48380") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115932 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26637 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101628 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66296 "Found 137 new API test failures: /WPE/TestWebsiteData:/webkit/WebKitWebsiteData/configuration, /WPE/TestBackForwardList:/webkit/WebKitWebView/navigation-after-session-restore, /WPE/TestInputMethodContext:/webkit/WebKitInputMethodContext/reset, /WPE/TestInputMethodContext:/webkit/WebKitInputMethodContext/surrounding, /WPE/TestLoaderClient:/webkit/WebKitURIResponse/http-headers, /WPE/TestAuthentication:/webkit/Authentication/authentication-ephemeral, /WPE/TestWebKitWebContext:/webkit/WebKitWebContext/timezone, /WPE/TestWebKitWebContext:/webkit/WebKitWebContext/timezone-worker, /WPE/TestWebKitUserContentManager:/webkit/WebKitUserContentManager/injected-script, /WPE/TestLoaderClient:/webkit/WebKitWebView/unfinished-subresource-load ... (failure)") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25916 "Found 4 new test failures: imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/computed-keyframes-shorthands.html (failure)") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19761 "Found 4 new test failures: imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/computed-keyframes-shorthands.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62947 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96025 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19835 "Found 4 new test failures: imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/computed-keyframes-shorthands.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122410 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40063 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29869 "Found 4 new test failures: imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/computed-keyframes-shorthands.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94838 "Found 4 new test failures: imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/computed-keyframes-shorthands.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40447 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97850 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94580 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39714 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17524 "Found 4 new test failures: imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001.html imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/computed-keyframes-shorthands.html (failure)") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36176 "Failed to checkout and rebase branch from PR 48380") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39949 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45448 "Failed to checkout and rebase branch from PR 48380") | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39590 "Failed to checkout and rebase branch from PR 48380") | | | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42923 "Failed to checkout and rebase branch from PR 48380") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41327 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/style/values/rhythm/StyleBlockStepSize.h:
I don't really have a strong preference, but any reason not to use LengthWrapperBase ?

weinig replied:
It models ‘<length-percentage>’ and keywords only at the moment. This is ‘<length>’. 

In the future, if I generalize it further, we can swap it in.

weinig committed:
[Style] Convert block-step-size property to use strong style types
https://bugs.webkit.org/show_bug.cgi?id=296335

Reviewed by NOBODY (OOPS!).

Convert `block-step-size` property to a strong style type.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/rendering/RenderBlock.cpp:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleExtractorSerializer.h:
* Source/WebCore/style/values/rhythm/StyleBlockStepSize.cpp: Added.
* Source/WebCore/style/values/rhythm/StyleBlockStepSize.h: Added.
